exchanges/gateio/gateio.go

```diff
@@ -1323,6 +1324,31 @@ func (e *Exchange) ConvertSmallBalances(ctx context.Context, currs ...currency.C
 
 // ********************************* Margin *******************************************
 
+// GetEstimatedInterestRate retrieves estimated interest rate for provided currencies
+func (e *Exchange) GetEstimatedInterestRate(ctx context.Context, currencies []currency.Code) (map[string]types.Number, error) {
+	if len(currencies) == 0 {
+		return nil, currency.ErrCurrencyCodesEmpty
+	}
+	if len(currencies) > 10 {
+		return nil, fmt.Errorf("%w: maximum 10", errTooManyCurrencyCodes)
+	}
+	var currStr strings.Builder
+	for i := range currencies {
+		if currencies[i].IsEmpty() {
+			return nil, currency.ErrCurrencyCodeEmpty
+		}
+		if i != 0 {
+			currStr.WriteString(",")
+		}
+		currStr.WriteString(currencies[i].String())
+	}
+	params := url.Values{}
+	params.Set("currencies", currStr.String())
+
+	var response map[string]types.Number
+	return response, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, publicCurrencyPairsMarginEPL, http.MethodGet, "margin/uni/estimate_rate", params, nil, &response)
```

Comment:
Since this is using auth and would place it under "other private endpoints" which would need otherPrivateEndpointRateLimit (150r/10s)

Reply:
Done: https://github.com/thrasher-corp/gocryptotrader/pull/2102/commits/8f22f3485660823d6ff3905978586a80ec948079